ResearchKit/Common/ORKFileImportStepView.m

```diff
@@ -0,0 +1,214 @@
+ are permitted provided that the following conditions are met:
+
+ 1.  Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer.
+
+ 2.  Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation and/or
+ other materials provided with the distribution.
+
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "ORKFileImportStepView.h"
+
+#import "ORKNavigationContainerView_Internal.h"
+
+#import "ORKVerticalContainerView_Internal.h"
+
+#import "ORKStepHeaderView_Internal.h"
+
+#import "ORKFileImportStep.h"
+#import "ORKHelpers_Internal.h"
+
+@interface PDFView : UIView
+
+@property (nullable, nonatomic, copy) NSURL *url;
+
+@end
+
+@implementation PDFView {
+
+}
+
+- (instancetype)initWithFrame:(CGRect)frame url:(nullable NSURL *)url
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        self.url = url;
+    }
+    return self;
+}
+
+- (void)drawRect:(CGRect)rect
+{
+    CGContextRef ctx = UIGraphicsGetCurrentContext();
+
+    // PDF might be transparent, assume white paper
+    [[UIColor lightGrayColor] set];
+    CGContextFillRect(ctx, rect);
+
+    if (!self.url) { return; }
+
+    // Flip coordinates
+    CGContextGetCTM(ctx);
+    CGContextScaleCTM(ctx, 1, -1);
+    CGContextTranslateCTM(ctx, 0, -rect.size.height);
+
+    // url is a file URL
+    CGPDFDocumentRef pdf = CGPDFDocumentCreateWithURL((CFURLRef)self.url);
+    CGPDFPageRef page1 = CGPDFDocumentGetPage(pdf, 1);
+
+    // get the rectangle of the cropped inside
+    CGRect mediaRect = CGPDFPageGetBoxRect(page1, kCGPDFCropBox);
+
+    [[UIColor whiteColor] set];
+
+    CGFloat border = 8;
+    CGFloat w = rect.size.width - 2*border;
+    CGFloat h = rect.size.height - 2*border;
+
+    CGFloat scale = MIN(w / mediaRect.size.width, h / mediaRect.size.height);
```

Comment:
💡 This is a very rudimentary PDF view - only first page, and it doesn't center correctly. 😞 good enough for now.